### PR TITLE
Remove connect delay

### PR DESCRIPTION
**Pull request by wallacbe**

### CODE REVIEW

**Comment by charlesnicholson** [comment]
I suggest unit tests but support this change...

File: pylink/jlink.py [hunk 1]
```diff
@@ -927,7 +928,7 @@ def connect(self, chip_name, speed='auto', verbose=False):
         # Required in the event that the device is unsecured, as it will be
         # mass erased here if the system is a Windows system and the user has
         # opted to erase the device.
-        time.sleep(2)
+        time.sleep(wait_after_connect_sec)
```
Comment (sstallion) [inline pylink/jlink.py]: I suspect adding a load bearing delay here isn't the right solution. @hkpeprah do we have a method of polling the host or target for when this operation might complete? It seems like that might be a cleaner solution than forcing callers to pull a number out of their rear for every target they support.

Comment (wallacbe) [reply pylink/jlink.py]: Added unit test

Comment (hkpeprah) [reply pylink/jlink.py]: Yah, the load bearing sleep probably isn't the right thing to do here.  We probably want to remove it altogether.  I don't believe we need to poll the host/target, just issue any no-op command that would block.

Comment (wallacbe) [reply pylink/jlink.py]: Is the current command after this suitable?  self._dll.JLINKARM_DEVICE_GetInfo(-1, 0)

Comment (hkpeprah) [reply pylink/jlink.py]: Unfortunately not.  That information is internal, so it doesn't talk to the target to get it.

Comment (sstallion) [reply pylink/jlink.py]: @hkpeprah What would be the ideal command?

Comment (hkpeprah) [reply pylink/jlink.py]: Hm.  Maybe querying if the CPU is halted via `.halted()`?  Would have to wrap it in a try-catch, but that should block.  We wouldn't/shouldn't have to care about the return value.

Comment (wallacbe) [reply pylink/jlink.py]: PTAL, thanks

**Comment by hkpeprah** [comment]
Tested, seems to be solid.  If you haven't signed the [CLA](https://docs.google.com/forms/d/e/1FAIpQLSeRVQ35-gq2vdSxD1kdh7CJwRdjmUA0EZ9gRXaWYoUeKPZEQQ/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1) could you sign that before the merge?

**Comment by wallacbe** [comment]
CLA done, Thanks

**Comment by hkpeprah** [comment]
I'll cut and release a new revision with these changes.

**Comment by wallacbe** [comment]
Thanks!

**Comment by hkpeprah** [comment]
Done

**Pull request merged**